packages/dx-grid-core/src/plugins/paging-state/computeds.js

```diff
@@ -1,8 +1,16 @@
-export const paginate = (rows, pageSize, page) =>
-  rows.slice(pageSize * page, pageSize * (page + 1));
+const showAllPagesKey = 'all';
+
+export const paginate = (rows, pageSize, page) => (
+  pageSize !== showAllPagesKey ?
+    rows.slice(pageSize * page, pageSize * (page + 1)) :
+    rows
+);
 
 export const ensurePageHeaders = (rows, pageSize) => {
   const result = rows.slice();
+  if (pageSize === showAllPagesKey) {
```

Comment:
This expression is used four times in this file. May be create a helper function?